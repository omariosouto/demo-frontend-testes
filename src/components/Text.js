import styled from 'styled-components';

const StyledText = styled.span`
    font-family: sans-serif;
`;

export function Text({ tag, children }) {
    return (
        <StyledText as={tag}>
            {children}
        </StyledText>
    )
} 