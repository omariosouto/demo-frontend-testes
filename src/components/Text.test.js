import { render } from '../utils/test/react-testing-library';
import { Text } from './Text';


describe('<Text />', () => {
    it('should render a <span> tag', () => {
        const { container } = render(<Text>Hello World</Text>);
        expect(container.firstChild.tagName).toBe('SPAN');
        expect(container).toMatchSnapshot();
    });
})