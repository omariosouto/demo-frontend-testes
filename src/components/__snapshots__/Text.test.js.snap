// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render a <span> tag 1`] = `
.c0 {
  font-family: sans-serif;
}

<div>
  <span
    class="c0"
  >
    Hello World
  </span>
</div>
`;
